docs/APP-SNAPSHOT.md

```diff
@@ -0,0 +1,1053 @@
+   system reboots. Without serialization, the snapshot information would be lost
+   from memory upon a system restart, making it impossible to perform operations
+   like rollback or deletion of the snapshot after a reboot.
+7. **Publishing the Snapshot Status**: After the snapshot has been successfully
+   created and the snapshot status has been serialized,
+   the `publishVolumesSnapshotStatus` function is called to publish
+   the `VolumesSnapshotStatus` object. By publishing the snapshot status,
+   volumemgr notifies zedmanager about the result of the snapshot creation
+   request. zedmanager can then use this information to update the application's
+   status and take any necessary actions based on the snapshot status.
+
+#### Zedmanager Reaction, creation
+
+1. **Receiving the Snapshot Status and Handling Snapshot Creation**: When
+   zedmanager receives a `volumesSnapshotStatus` message, it calls
+   the `handleVolumesSnapshotStatusImpl` function. The function examines
+   the `ResultOfAction` field of the `volumesSnapshotStatus` object to determine
+   the type of action that was performed (e.g., snapshot creation, rollback, or
+   deletion). If the `ResultOfAction` field indicates that a snapshot was
+   created, the `reactToSnapshotCreate` function is called to handle the
+   creation of the snapshot.
+2. **Looking Up AppInstanceStatus**: The `reactToSnapshotCreate` function begins
+   by looking up the `AppInstanceStatus` object associated with the `AppUUID`
+   field of the `volumesSnapshotStatus` object. If the `AppInstanceStatus`
+   object is not found, an error is logged, and the function returns.
+   The `AppInstanceStatus` object is necessary for updating the status of the
+   snapshot, which is done in the next step.
+3. **Moving Snapshot to Available**: If there is no error, the function logs
+   that the snapshot has been created and calls the `moveSnapshotToAvailable`
+   function. This function removes the snapshot from the `RequestedSnapshots`
+   slice of the `SnapStatus` field of the `AppInstanceStatus` object and adds it
+   to the `AvailableSnapshots` slice. The function also updates
+   the `TimeCreated` field of the snapshot to the value from
+   the `volumesSnapshotStatus` object, marks the snapshot as reported, and
+   serializes the `SnapshotInstanceStatus` object to a file.
+4. **Publishing AppInstanceStatus**: The `AppInstanceStatus` object is then
+   published to reflect the changes made during the snapshot creation process.
+5. **Zedagent Handling**: After the publication of the `AppInstanceStatus`, the
+   updated status is sent to zedagent. Zedagent then parses the list of
+   available snapshots in the `AppInstanceStatus` and forms a reply to the
+   controller with the updated snapshot information. This allows the controller
+   to be aware of the current state of the snapshots on the device.
+
+### Snapshot Rollback Process
+
+#### Local Application Instance Config
```

Comment:
Might make sense to mention in this section that the local copy is serialized to persistent storage to make the solution handle device restarts at any time during the snapshot and rollback processes.

Reply:
will do